block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagStorage.scala

```diff
@@ -22,7 +22,7 @@ trait BlockDagRepresentation[F[_]] {
   def lookup(blockHash: BlockHash): F[Option[BlockMetadata]]
   def contains(blockHash: BlockHash): F[Boolean]
   def topoSort(startBlockNumber: Long): F[Vector[Vector[BlockHash]]]
-  def topoSortTail(tailLength: Long): F[Vector[Vector[BlockHash]]]
+  def topoSortTail(tailLength: Int): F[Vector[Vector[BlockHash]]]
```

Comment:
`tailLength` basically represents the length of the resulting vector. Since we can't create vectors longer than 2^31-1, it does not make sense to allow to specify a `Long` length for it. On the other hand, `startBlockNumber` just represents the offset which could potentially be larger than 2^31-1, but it is okay since the entire data structure won't be stored in memory.